README.md

```diff
@@ -416,6 +414,88 @@ Volume requests embedded in Pod spec are provisioned as ephemeral volumes. The v
+
+Work is [under
+way](https://github.com/kubernetes/enhancements/pull/1353) to enhance
+scheduling in Kubernetes. In the meantime, PMEM-CSI provides two components
+that help with pod scheduling:
+
+#### Scheduler extender
+
+When a pod requests the special [extended
+resource](https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/#extended-resources)
+called `pmem-csi.intel.com/scheduler`, the Kubernetes scheduler calls
+a [scheduler
+extender](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/scheduling/scheduler_extender.md)
+provided by PMEM-CSI with a list of nodes that a pod might run
+on. This extender is implemented in the master controller and thus can
+connect to the controller on each of these nodes to check for
+capacity. PMEM-CSI then filters out all nodes which currently do not
+have enough storage left for the volumes that still need to be
+created. This considers inline ephemeral volumes and all unbound
+volumes, regardless whether they use late binding or immediate
+binding.
+
+This special scheduling can be requested manually by adding this snippet
+to one container in the pod spec:
+```
+containers:
+- name: some-container
+  ...
+  resources:
+    limits:
+      pmem-csi.intel.com/scheduler: "1"
+    requests:
+      pmem-csi.intel.com/scheduler: "1"
+```
+
+This scheduler extender is optional and not necessarily installed in
+all clusters that have PMEM-CSI. Don't add this extended resource
+unless the scheduler extender is installed, otherwise the pod won't
+start!
+
+#### Pod admission webhook
+
+Having to add `pmem-csi.intel.com/scheduler` manually is not
+user-friendly. To simplify this, PMEM-CSI provides a [mutating
+admission
+webhook](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/)
+which intercepts the creation of all pods. If that pod uses volumes
+provided by PMEM-CSI, the webhook transparently adds the extended
+resource request.
+
+The webhook can only do that a) if the persistent volume claim (PVC)
+and its storage class have been created already and b) the webhook has
+been informed about them. The former is not guaranteed: it's okay to
+create the pod first, then later add the PVC. The pod simply won't
+start in the meantime. Even if these additional objects get created
+first, there is still a race condition because the webhook might get
+called before the controller gets a copy of the new storage class and
+PVC.
+
+The webhook deals with this uncertainty by allowing the creation of
+the pod without adding the extended resource when it lacks the
+necessary information. The alternative would be to reject the pod, but
+that is a change of behavior of the cluster that may affect also pods
+that don't use PMEM-CSI at all.
+
+Because of this limitation, it is more reliable to manually add the
+resource section.
+
```

Comment:
The above chapter about webhook: What is the final point I get? 
We have manual config, and to make it more friendly, we have webhook, which is not reliable? Why would I choose unreliable solution over user-unfriendly but reliable solution?
Is webhook feature present only to help with this not-user-friendly issue, but the cost is lower reliability?

Reply:
It's a tradeoff that users have to make. We can offer both, but we can't decide for them which approach they want to use. Neither one nor the other is perfect (one is user-friendly, the other reliable).

Reply:
I kept thinking about this. The unreliability stems from using an informer, which by design may be not up-to-date yet when it is queried. I think it is possible to add fallbacks to the slower direct get calls when the informer doesn't have an item, and then this can be reduced to "it'll not work when PVC/StorageClass are created after pod", which is something that users can avoid.